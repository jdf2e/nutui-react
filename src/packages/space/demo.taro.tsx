import React from 'react'
import Taro from '@tarojs/taro'
import { ScrollView, View } from '@tarojs/components'
import { Cell } from '@nutui/nutui-react-taro'
import { useTranslate } from '@/sites/assets/locale/taro'
import Header from '@/sites/components/header'
import Demo1 from './demos/taro/demo1'
import Demo2 from './demos/taro/demo2'
import Demo3 from './demos/taro/demo3'
// import Demo4 from './demos/taro/demo4'
import Demo5 from './demos/taro/demo5'
import Demo6 from './demos/taro/demo6'

const SpaceDemo = () => {
  const [translated] = useTranslate({
    'zh-CN': {
      basic: '基本用法',
      wrap: '换行',
      direction: '垂直',
      spaceGap: '间距大小',
      mainAxisAlign: '主轴对齐方式',
      crossAxisAlign: '交叉轴对齐方式',
    },
    'zh-TW': {
      basic: '基本用法',
      wrap: '換行',
      direction: '垂直',
      spaceGap: '間距大小',
      mainAxisAlign: '主軸對齊方式',
      crossAxisAlign: '交叉軸對齊方式',
    },
    'en-US': {
      basic: 'Basic Usage',
      wrap: 'wrap',
      direction: 'Direction',
      spaceGap: 'SpaceGap',
      mainAxisAlign: 'MainAxis Alignment',
      crossAxisAlign: 'CrossAxis Alignment',
    },
  })

  return (
    <>
      <Header />
      <ScrollView className={`demo ${Taro.getEnv() === 'WEB' ? 'web' : ''}`}>
        <View className="h2">{translated.basic}</View>
        <Cell>
          <Demo1 />
        </Cell>
        <View className="h2">{translated.wrap}</View>
        <Cell>
          <Demo2 />
        </Cell>
        <View className="h2">{translated.direction}</View>
        <Cell>
          <Demo3 />
        </Cell>
        <View className="h2">{translated.spaceGap}</View>
        {/* @TODO ConfigProvider 暂不支持 */}
        {/* <Cell>
          <Demo4 />
        </Cell> */}
        <View className="h2">{translated.mainAxisAlign}</View>
        <Cell style={{ display: 'block' }}>
          <Demo5 />
        </Cell>
        <View className="h2">{translated.crossAxisAlign}</View>
        <Cell>
          <Demo6 />
        </Cell>
      </ScrollView>
    </>
  )
}

export default SpaceDemo
