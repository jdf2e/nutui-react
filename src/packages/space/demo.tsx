import React from 'react'
import Cell from '@/packages/cell'
import { useTranslate } from '@/sites/assets/locale'
import Demo1 from './demos/h5/demo1'
import Demo2 from './demos/h5/demo2'
import Demo3 from './demos/h5/demo3'
import Demo4 from './demos/h5/demo4'
import Demo5 from './demos/h5/demo5'
import Demo6 from './demos/h5/demo6'

const SpaceDemo = () => {
  const [translated] = useTranslate({
    'zh-CN': {
      basic: '基本用法',
      wrap: '换行',
      direction: '垂直',
      spaceGap: '间距大小',
      mainAxisAlign: '主轴对齐方式',
      crossAxisAlign: '交叉轴对齐方式',
    },
    'zh-TW': {
      basic: '基本用法',
      wrap: '換行',
      direction: '垂直',
      spaceGap: '間距大小',
      mainAxisAlign: '主軸對齊方式',
      crossAxisAlign: '交叉軸對齊方式',
    },
    'en-US': {
      basic: 'Basic Usage',
      wrap: 'wrap',
      direction: 'Direction',
      spaceGap: 'SpaceGap',
      mainAxisAlign: 'MainAxis Alignment',
      crossAxisAlign: 'CrossAxis Alignment',
    },
  })

  return (
    <>
      <div className="demo">
        <h2>{translated.basic}</h2>
        <Cell>
          <Demo1 />
        </Cell>
        <h2>{translated.wrap}</h2>
        <Cell>
          <Demo2 />
        </Cell>
        <h2>{translated.direction}</h2>
        <Cell>
          <Demo3 />
        </Cell>
        <h2>{translated.spaceGap}</h2>
        <Cell>
          <Demo4 />
        </Cell>
        <h2>{translated.mainAxisAlign}</h2>
        <Cell style={{ display: 'block' }}>
          <Demo5 />
        </Cell>
        <h2>{translated.crossAxisAlign}</h2>
        <Cell>
          <Demo6 />
        </Cell>
      </div>
    </>
  )
}

export default SpaceDemo
