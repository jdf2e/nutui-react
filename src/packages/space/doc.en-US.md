# Space



Maintain the same width in the arrangement of elements.

## Import

```tsx
import { Space } from '@nutui/nutui-react';
```

## Code

### Basic Usage

:::demo

<CodeBlock src='h5/demo1.tsx'></CodeBlock>

:::

### Wrap

:::demo

<CodeBlock src='h5/demo2.tsx'></CodeBlock>

:::

### Direction

:::demo

<CodeBlock src='h5/demo3.tsx'></CodeBlock>

:::

### Gap

:::demo

<CodeBlock src='h5/demo4.tsx'></CodeBlock>

:::

### MainAxis alignment

:::demo

<CodeBlock src='h5/demo5.tsx'></CodeBlock>

:::

### CrossAxis alignment

:::demo

<CodeBlock src='h5/demo6.tsx'></CodeBlock>

:::

## Space

### Props

| Property | Description | Type | Default |
| --- | --- | --- | --- |
| direction | space direction | `vertical \| horizontal` | `horizontal` |
| align | space align | `start \| end \|center \| baseline` | `-` |
| justify | space justify | `start \| end \| center \| between \| around \| evenly \| stretch` | `-` |
| wrap | space wrapï¼ŒOnly valid when horizontal | `boolean` | `false` |

## Theming

### CSS Variables

The component provides the following CSS variables, which can be used to customize styles. Please refer
to [ConfigProvider component](#/en-US/component/configprovider).

| Name | Description | Default           |
| --- | --- |--------------|
| \--nutui-space-gap | `8px` | spacing size |
