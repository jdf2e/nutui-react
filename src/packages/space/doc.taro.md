# Space组件

### 介绍

元素排列中保持相同的宽度。

### 安装

```tsx
import { Space } from '@nutui/nutui-react-taro';
```

## 代码演示

### 基础用法

:::demo

```tsx

import React from 'react';
import { Space, Button } from '@nutui/nutui-react-taro';

 const App = () => {
  return (
    <Space>
      <Button>按钮1</Button>
      <Button>按钮2</Button>
      <Button>按钮3</Button>
    </Space>
  );
};
export default App;

```

### 换行

:::demo

```tsx
import React from 'react';
import { Space, Button } from '@nutui/nutui-react-taro';

const App = () => {
  return (
    <Space wrap>
      <Button>按钮1</Button>
      <Button>按钮2</Button>
      <Button>按钮3</Button>
      <Button>按钮4</Button>
      <Button>按钮5</Button>
      <Button>按钮6</Button>
    </Space>
  );
};
export default App;

```

### 垂直

:::demo

```tsx
import React from 'react';
import { Space, Button } from '@nutui/nutui-react-taro';

const App = () => {
  return (
    <Space direction="vertical">
      <Button>按钮1</Button>
      <Button>按钮2</Button>
      <Button>按钮3</Button>
    </Space>
  );
};
export default App;

```

### 间距大小

:::demo

```tsx
import React from 'react';
import { Space, Button,ConfigProvider } from '@nutui/nutui-react-taro';

const App = () => {
  return (
    <ConfigProvider
      theme={{
        nutuiSpaceGap: '20px',
      }}
    >
      <Space direction="vertical">
        <Button>按钮1</Button>
        <Button>按钮2</Button>
        <Button>按钮3</Button>
      </Space>
    </ConfigProvider>
  );
};
export default App;

```

## Space

### Props

| 属性 | 描述 | 类型 | 默认值 |
| --- | --- | --- | --- |
| direction | 间距方向 | `'vertical'｜ 'horizontal'` | `'horizontal'` |
| align | 交叉轴对齐方式 | `'start'｜'end'｜'center'｜'baseline'` | `-` |
| justify | 主轴对齐方式 | `'start' ｜ 'end' ｜ 'center' ｜ 'between' ｜ 'around' ｜ 'evenly' ｜ 'stretch'` | `-` |
| wrap | 是否自动换行，仅在 horizontal 时有效 | `boolean / false` | `-` |

## 主题定制

### 样式变量

组件提供了下列 CSS 变量，可用于自定义样式，使用方法请参考 [ConfigProvider 组件](/components/config-provider)。

| 名称 | 默认值 | 描述 |
| --- | --- | --- |
| \--nutui-space-gap | `8px` | 间距大小 |