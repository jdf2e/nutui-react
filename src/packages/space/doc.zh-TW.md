# Space 間距

### 介紹

元素排列中保持相同的寬度。

### 安裝

```tsx
import { Space } from '@nutui/nutui-react';
```

## 代碼演示

### 基礎用法

:::demo

```tsx

import React from 'react';
import { Space, Button } from '@nutui/nutui-react';

 const App = () => {
  return (
    <Space>
      <Button>按鈕1</Button>
      <Button>按钮2</Button>
      <Button>按钮3</Button>
    </Space>
  );
};
export default App;

```

:::

### 換行

:::demo

```tsx
import React from 'react';
import { Space, Button } from '@nutui/nutui-react';

const App = () => {
  return (
    <Space wrap>
      <Button>按鈕1</Button>
      <Button>按钮2</Button>
      <Button>按钮3</Button>
      <Button>按钮4</Button>
      <Button>按钮5</Button>
      <Button>按钮6</Button>
    </Space>
  );
};
export default App;

```

:::

### 垂直

:::demo

```tsx
import React from 'react';
import { Space, Button } from '@nutui/nutui-react';

const App = () => {
  return (
    <Space direction="vertical">
      <Button>按鈕1</Button>
      <Button>按钮2</Button>
      <Button>按钮3</Button>
    </Space>
  );
};
export default App;

```

:::

### 間距大小

:::demo

```tsx
import React from 'react';
import { Space, Button,ConfigProvider } from '@nutui/nutui-react';

const App = () => {
  return (
    <ConfigProvider
      theme={{
        nutuiSpaceGap: '20px',
      }}
    >
      <Space direction="vertical">
        <Button>按鈕1</Button>
        <Button>按钮2</Button>
        <Button>按钮3</Button>
      </Space>
    </ConfigProvider>
  );
};
export default App;

```

### 主軸對齊方式

:::demo

```tsx
import React from 'react';
import { Space, Button } from '@nutui/nutui-react';

const App = () => {
  return (
    <Space justify="center" wrap>
      <Button style={{ height: '30px' }}>按钮1</Button>
      <Button style={{ height: '60px' }}>按钮2</Button>
      <Button style={{ height: '90px' }}>按钮3</Button>
    </Space>    
  );
};
export default App;

```

### 交叉軸對齊方式

:::demo

```tsx
import React from 'react';
import { Space, Button } from '@nutui/nutui-react';

const App = () => {
  return (
    <Space align="end" wrap>
      <Button style={{ height: '30px' }}>按钮1</Button>
      <Button style={{ height: '60px' }}>按钮2</Button>
      <Button style={{ height: '90px' }}>按钮3</Button>
    </Space>  
  );
};
export default App;

```

:::

## Space

### Props

| 屬性 | 說明 | 類型 | 默認值 |
| --- | --- | --- | --- |
| direction | 間距方嚮 | `'vertical'｜ 'horizontal'` | `'horizontal'` |
| align | 交叉軸對齊方式 | `'start'｜'end'｜'center'｜'baseline'` | `-` |
| justify | 主軸對齊方式 | `'start' ｜ 'end' ｜ 'center' ｜ 'between' ｜ 'around' ｜ 'evenly' ｜ 'stretch'` | `-` |
| wrap | 是否自動換行，僅在 horizontal 時有效 | `boolean` | `false` |

## 主題定制

### 樣式變量

組件提供了下列 CSS 變量，可用於自定義樣式，使用方法請參考 [ConfigProvider 組件](#/zh-CN/component/configprovider)。

| 名稱 | 默認值 | 說明 |
| --- | --- | --- |
| \--nutui-space-gap | `8px` | 间距大小 |
