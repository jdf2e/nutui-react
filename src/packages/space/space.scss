.nut-space {
  display: flex;

  &-item {
    flex: none;
  }

  &-vertical {
    flex-direction: column;

    > .nut-space-item {
      margin-block-end: $space-gap;

      &:last-child {
        margin-block-end: 0;
      }
    }
  }

  &-horizontal {
    flex-direction: row;

    > .nut-space-item {
      margin-inline-end: $space-gap;

      &:last-child {
        margin-inline-end: 0;
      }
    }

    &.nut-space-wrap {
      flex-wrap: wrap;
      margin-block-end: calc(#{$space-gap} * -1);

      > .nut-space-item {
        padding-block-end: $space-gap;
      }
    }
  }

  &-align {
    &-center {
      align-items: center;
    }

    &-start {
      align-items: flex-start;
    }

    &-end {
      align-items: flex-end;
    }

    &-baseline {
      align-items: baseline;
    }
  }

  &-justify {
    &-center {
      justify-content: center;
    }

    &-start {
      justify-content: flex-start;
    }

    &-end {
      justify-content: flex-end;
    }

    &-between {
      justify-content: space-between;
    }

    &-around {
      justify-content: space-around;
    }

    &-evenly {
      justify-content: space-evenly;
    }

    &-stretch {
      justify-content: stretch;
    }
  }
}
