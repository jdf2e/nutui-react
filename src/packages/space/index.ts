import { Space } from './space'

export default Space
