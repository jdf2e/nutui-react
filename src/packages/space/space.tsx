import React, { FunctionComponent } from 'react'
import classNames from 'classnames'
import { useConfig } from '@/packages/configprovider'
import { BasicComponent } from '@/utils/typings'

const prefixCls = 'nut-space'

export interface SpaceProps extends BasicComponent {
  direction: 'horizontal' | 'vertical'
  align: 'start' | 'end' | 'center' | 'baseline'
  justify:
    | 'start'
    | 'end'
    | 'center'
    | 'between'
    | 'around'
    | 'evenly'
    | 'stretch'
  wrap: boolean
  onClick: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void
}
const defaultProps = {
  direction: 'horizontal',
} as SpaceProps

export const Space: FunctionComponent<
  Partial<SpaceProps> & React.HTMLAttributes<HTMLDivElement>
> = (props) => {
  const { locale } = useConfig()
  const { children, onClick, wrap, align, direction, justify } = {
    ...defaultProps,
    ...props,
  }
  const cls = classNames(
    prefixCls,
    wrap && `${prefixCls}-wrap`,
    direction && `${prefixCls}-${direction}`,
    align && `${prefixCls}-align-${align}`,
    justify && `${prefixCls}-justify-${justify}`
  )
  return (
    <div className={cls}>
      {React.Children.map(children, (child) => {
        return (
          child !== null &&
          child !== undefined && (
            <div className={`${prefixCls}-item`}>{child}</div>
          )
        )
      })}
    </div>
  )
}

Space.defaultProps = defaultProps
Space.displayName = 'NutSpace'
