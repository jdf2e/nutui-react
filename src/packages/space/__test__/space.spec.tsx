import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import { Space } from '../space'
import Button from '@/packages/button'

const prefixCls = 'nut-space'

test('should match snapshot', () => {
  const { container } = render(
    <Space>
      <Button>按钮1</Button>
      <Button>按钮2</Button>
      <Button>按钮3</Button>
    </Space>
  )
  expect(container).toMatchSnapshot()
})

test('render with props direction', () => {
  const { container } = render(
    <Space direction="vertical">
      <Button>按钮1</Button>
      <Button>按钮2</Button>
      <Button>按钮3</Button>
    </Space>
  )
  const nutSpace = container.getElementsByClassName('nut-space')[0]
  expect(nutSpace).toHaveClass(`${prefixCls}-vertical`)
})

test('render with props align', () => {
  const { container } = render(
    <Space align="end">
      <Button>按钮1</Button>
      <Button>按钮2</Button>
      <Button>按钮3</Button>
    </Space>
  )
  const nutSpace = container.getElementsByClassName('nut-space')[0]
  expect(nutSpace).toHaveClass(`${prefixCls}-align-end`)
})

test('render with props justify', () => {
  const { container } = render(
    <Space justify="center">
      <Button>按钮1</Button>
      <Button>按钮2</Button>
      <Button>按钮3</Button>
    </Space>
  )
  const nutSpace = container.getElementsByClassName('nut-space')[0]
  expect(nutSpace).toHaveClass(`${prefixCls}-justify-center`)
})

test('render with props wrap', () => {
  const { container } = render(
    <Space wrap>
      <Button>按钮1</Button>
      <Button>按钮2</Button>
      <Button>按钮3</Button>
      <Button>按钮4</Button>
      <Button>按钮5</Button>
      <Button>按钮6</Button>
    </Space>
  )
  const nutSpace = container.getElementsByClassName('nut-space')[0]
  expect(nutSpace).toHaveClass(
    `nut-space nut-space-horizontal nut-space-horizontal-wrap`
  )
})
