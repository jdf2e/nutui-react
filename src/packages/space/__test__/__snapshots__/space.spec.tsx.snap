// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="nut-space nut-space-horizontal"
  >
    <div
      class="nut-space-item"
    >
      <button
        class="nut-button nut-button-default nut-button-normal nut-button-round"
        type="button"
      >
        <div
          class="nut-button-wrap"
        >
          <div
            class=""
          >
            按钮1
          </div>
        </div>
      </button>
    </div>
    <div
      class="nut-space-item"
    >
      <button
        class="nut-button nut-button-default nut-button-normal nut-button-round"
        type="button"
      >
        <div
          class="nut-button-wrap"
        >
          <div
            class=""
          >
            按钮2
          </div>
        </div>
      </button>
    </div>
    <div
      class="nut-space-item"
    >
      <button
        class="nut-button nut-button-default nut-button-normal nut-button-round"
        type="button"
      >
        <div
          class="nut-button-wrap"
        >
          <div
            class=""
          >
            按钮3
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;
